### PR TITLE
[packaging iOS/tvOS] fix wrong architecture field in deb control file

### PR DESCRIPTION
**Pull request by sy6sy2**
## Description

This PR fix the wrong value of the `Architecture` field in the DEB control file for iOS and tvOS.

## Motivation and Context

Since the initial tvOS PR (I guess) all iOS and tvOS nightly builds debs files can not be installed on jailbroken devices anymore because the `Architecture` field value in the DEB control file is wrong.

On iOS you get this error:
```shell
iPhone-de-Sylvain:/var/mobile/Downloads root# dpkg -i kodi-20200418-0e954dea-master-ios64.deb      
dpkg: error processing archive kodi-20200418-0e954dea-master-ios64.deb (--install):                
 package architecture (Darwin-arm64) does not match system (iphoneos-arm)                          
Errors were encountered while processing:        
 kodi-20200418-0e954dea-master-ios64.deb
```

And on tvOS you get (source: https://forum.kodi.tv/showthread.php?tid=353675&pid=2942310#pid2942310):
```shell
root@ (/var/root)# dpkg -i kodi-20200418-0e954dea-master-tvos.deb 
dpkg: error processing archive kodi-20200418-0e954dea-master-tvos.deb (--install):
 package architecture (Darwin-arm64) does not match system (darwin-arm64)
Errors were encountered while processing:
 kodi-20200418-0e954dea-master-tvos.deb
```

## How Has This Been Tested?

I can now install the deb file on my jailbroken iPhone with the `dpkg` command.
But I don't have a jailbroken Apple TV to test on this platform.
But according to Memphiz yab branch, it should works --> https://github.com/Memphiz/xbmc/blob/968cd92bcb210bd95cfbfcc86a5cff02109e4b26/tools/darwin/packaging/tvos/mkdeb-tvos.sh.in#L73

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by Rechi** [comment]
jenkins build this please

File: tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in [hunk 1]
```diff
@@ -75,7 +75,11 @@ echo "Conflicts: $PACKAGE"                      >> $DIRNAME/$PACKAGE/DEBIAN/cont
 echo "Replaces: $PACKAGE"                       >> $DIRNAME/$PACKAGE/DEBIAN/control
 echo "Priority: Extra"                            >> $DIRNAME/$PACKAGE/DEBIAN/control
 echo "Version: $VERSION-$REVISION"                >> $DIRNAME/$PACKAGE/DEBIAN/control
-echo "Architecture: @CMAKE_SYSTEM_NAME@-@CPU@"                 >> $DIRNAME/$PACKAGE/DEBIAN/control
+if [ "@PLATFORM@" == "iphoneos" ]; then
+  echo "Architecture: iphoneos-arm"                 >> $DIRNAME/$PACKAGE/DEBIAN/control
+else
+  echo "Architecture: appletvos-arm64"                 >> $DIRNAME/$PACKAGE/DEBIAN/control
```
Comment (fuzzard) [inline tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: According to the snippet in your PR description, shouldn't this be darwin-arm64?

Comment (SylvainCecchetto) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Good question... I am still not sure between Darwin and appletvos.
According to Memphiz yab branch code it should be appletvos, but the user error shows darwin.
Anybody here have a jailbroken Apple TV to try both?

Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Does @PLATFORM@ could the correct platform name?

Comment (SylvainCecchetto) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Yes, but we also need to use `arm` for iOS and `arm64` for tvOS :-/
What do you mean by "the deb names"?

Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Example filename ms of the generated .deb files

Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: And `iphoneos-arm64` won’t work? Pity

Comment (SylvainCecchetto) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: I don't is there is any convention, on Cydia, for example, the NewTerm2 app us this filename: `ws.hang.newterm2_2.4_iphoneos-arm.deb`. I can look for others examples if you want.

I will try tomorrow with `iPhones-arm64`.

Comment (SylvainCecchetto) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: So, results of my test on iPhone 7:
* `iphoneos-arm`: `dpkg` is happy and install Kodi correctly
* `iphoneos-arm64`: `dpkg` error (package architecture (xxxxx) does not match system (iphoneos-arm))
* `darwin-arm`: idem
* `darwin-arm64`: idem

Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: What does `idem` mean?

Comment (SylvainCecchetto) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Oh sorry, I thought it was international. "Idem" means "same result as above".

File: tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in [hunk 2]
```diff
@@ -75,7 +75,11 @@ echo "Conflicts: $PACKAGE"                      >> $DIRNAME/$PACKAGE/DEBIAN/cont
 echo "Replaces: $PACKAGE"                       >> $DIRNAME/$PACKAGE/DEBIAN/control
 echo "Priority: Extra"                            >> $DIRNAME/$PACKAGE/DEBIAN/control
 echo "Version: $VERSION-$REVISION"                >> $DIRNAME/$PACKAGE/DEBIAN/control
-echo "Architecture: @CMAKE_SYSTEM_NAME@-@CPU@"                 >> $DIRNAME/$PACKAGE/DEBIAN/control
+if [ "@PLATFORM@" == "iphoneos" ]; then
+  echo "Architecture: iphoneos-arm"                 >> $DIRNAME/$PACKAGE/DEBIAN/control
```
Comment (phunkyfish) [inline tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:79]: Maybe `@PLATFORM@-arm`

File: tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in (hunk 1 above)
Comment (phunkyfish) [inline tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Maybe `@PLATFORM@-arm64`

Do we have a user with jailbroken apple tv? Would be good to know that this is correct.

Comment (SylvainCecchetto) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Yes we have senad.aruc on this thread https://forum.kodi.tv/showthread.php?tid=353675&page=2.
I will send him a patched tvOS deb file with `appletvos-arm64` to be sure that this is the correct value to use.

Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Nope, well not that I’ve heard 😉

Comment (SylvainCecchetto) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: It must be French then 😋

Comment (SylvainCecchetto) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: * According to the user, with `appletvos-arm64` the install is ok with dpkg (but it seems that the user has another during the runtime).
* And with `darwin-arm64` there is an error --> `dpkg: regarding kodi-20200421-9fab130c-master-tvos_control_file_patched_darwin_arm64.deb containing org.xbmc.kodi-tvos64, pre-dependency problem:  org.xbmc.kodi-tvos64:appletvos-arm64 pre-depends on cy+cpu.arm64`

So, aside the runtime error, `appletvos-arm64` seems to be the good one (same as Memphiz yab branch)

https://forum.kodi.tv/showthread.php?tid=353675&pid=2942955#pid2942955

Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Cool, so the PR is good as it is now.

Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Unless you want to update the literals to `@PLATFORM@` where possible.

Comment (SylvainCecchetto) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Even if we use `@PLATFROM@` the if/else remains. So I vote to keep it with iphoneos and appletvos string. But if one of you want to use the variable I can change it 😉

Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: Up to you guys. I’m easy either way but I generally go for less string literals if possible.

Comment (fuzzard) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: In this case I'd probably lean towards the explicit setting. Don't want to have to deal with this if something in Kodi changes. This is very specific to a single use case (jailbroken dpkg).

Comment (phunkyfish) [reply tools/darwin/packaging/darwin_embedded/mkdeb-darwin_embedded.sh.in:81]: 👍

**Review by phunkyfish — APPROVED** [review]
Then we are good 😉

**Pull request merged**